server/handles/fsread.go

```diff
@@ -260,16 +265,23 @@ func FsGet(c *gin.Context) {
 			return
 		}
 		if storage.Config().MustProxy() || storage.GetStorage().WebProxy {
+			query := "?"
+			if needSign(meta, reqPath) {
+				query = query + "sign=" + sign.Sign(reqPath)
+			}
+			if query == "?" {
+				query = ""
+			}
 			if storage.GetStorage().DownProxyUrl != "" {
```

Comment:
For `DownProxyUrl`, sign param is required regardless of whether sign is turned on in the settings due to its implementation.